### PR TITLE
added comment_count in tests, new model commentsByArticle

### PR DESCRIPTION
**Pull request by piknik1979**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ describe("1. GET /api/topics", () => {
   });
 });
 describe("GET /api/articles/:article_id", () => {
-  test("status:200, responds with single matching article object", () => {
+  test.only("status:200, responds with single matching article object", () => {
```
Comment (P-Copley) [inline __tests__/app.test.js]: Be careful of commiting these `.onlys` it's an easy way to skip important tests and introduce bugs by accident.

**Comment by P-Copley** [comment]
Doing this in 2 queries as you've done totally works but it's not necessary in this case. 

Consider using some sort of `JOIN` to do this in a single query as this will perform significantly better when dealing with large amounts of data.

File: __tests__/app.test.js (hunk 1 above)
Comment (piknik1979) [reply __tests__/app.test.js]: I have it removed now Thanks for this, Paul! I need to remember about these little things :D 
Is I OK to merge it now to the main

**Comment by piknik1979** [comment]
> Doing this in 2 queries as you've done totally works but it's not necessary in this case.
> 
> Consider using some sort of `JOIN` to do this in a single query as this will perform significantly better when dealing with large amounts of data.

Sure, I will try to to this now

**Comment by piknik1979** [comment]
> Doing this in 2 queries as you've done totally works but it's not necessary in this case.
> 
> Consider using some sort of `JOIN` to do this in a single query as this will perform significantly better when dealing with large amounts of data.

I have done few changes , using Join inside the models

File: controllers/news.controllers.js [hunk 2]
```diff
@@ -15,9 +16,16 @@ exports.getTopics = (req, res) => {
...
 exports.getArticleById = (req, res, next) => {
   const { article_id } = req.params;
-  selectArticleById(article_id)
-    .then((article) => {
-      res.status(200).send({ article: article });
+  const promises = [
+    selectArticleById(article_id),
+    commentsByArticle(article_id),
+  ];
+  Promise.all(promises)
```
Comment (jadelandeg) [inline controllers/news.controllers.js]: should be getting this information with one query rather than a promise.all

**Review by jadelandeg — CHANGES_REQUESTED** [review]
Good work, however should be using a single query to get all of this information. Also shouldn't be changing old tests, should add new test that asserts the new property with the correct value. To do this you can use the objectContaining matcher

File: __tests__/app.test.js [hunk 3]
```diff
@@ -54,9 +54,19 @@ describe("GET /api/articles/:article_id", () => {
           title: "A",
           topic: "mitch",
           votes: 0,
+          comment_count: 1,
         });
       });
   });
+  test.only("status:200, matching article object with comment_count", () => {
```
Comment (karlaferry) [inline __tests__/app.test.js]: No need for a separate test to check the number of comment_count separately. If you included comment_count on line 57 you should be fine without this!

File: controllers/news.controllers.js [hunk 4]
```diff
@@ -15,9 +16,13 @@ exports.getTopics = (req, res) => {
...
   const { article_id } = req.params;
-  selectArticleById(article_id)
-    .then((article) => {
-      res.status(200).send({ article: article });
+
+  Promise.all([selectArticleById(article_id), commentsByArticle(article_id)])
+    .then((results) => {
+      const article = results[0];
+      const { count } = results[1];
+      article.comment_count = Number(count);
```
Comment (karlaferry) [inline controllers/news.controllers.js]: You shouldn't make another query just to check the comment_count. Have a look at aggregate functions (hint: count) and using joins. You should be able to make only _one_ query in the models for this which will give you all the article information PLUS the comment_count.

**Review by karlaferry — CHANGES_REQUESTED** [review]
Please see my comments and make the necessary edits and resubmit for approval please. Thanks!

File: __tests__/app.test.js (hunk 3 above)
Comment (piknik1979) [reply __tests__/app.test.js]: Thank you! I see I think I misunderstood previous Jade comment...

File: __tests__/app.test.js [hunk 5]
```diff
@@ -177,7 +179,16 @@ describe("GET /api/users", () => {
       });
   });
 });
-// Responds with:
 
-// an array of objects, each object should have the following property:
-// username
+exports.selectArticles = () => {
```
Comment (karlaferry) [inline __tests__/app.test.js]: Can you please move this in the correct models file? You've put it in your test file 😅

File: controllers/news.controllers.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
+const comments = require("../db/data/test-data/comments");
```
Comment (karlaferry) [inline controllers/news.controllers.js]: Why are you requiring the test data here?

**Review by karlaferry — CHANGES_REQUESTED** [review]
Good work Tomek! Left a couple more comments about things you need to change. 😄

File: __tests__/app.test.js (hunk 5 above)
Comment (piknik1979) [reply __tests__/app.test.js]: I did now, I think I have been playing with this function. Early morning for me :)

File: controllers/news.controllers.js (hunk 6 above)
Comment (piknik1979) [reply controllers/news.controllers.js]: I don't know where it came from lol > I have it removed, now, Karla.

File: __tests__/app.test.js (hunk 5 above)
Comment (piknik1979) [reply __tests__/app.test.js]: It was in models file, actually, but I got it copied to the test I remember now so I can compare with what I had before, as well . But it sits on the models now only. Tests are working

**Pull request merged**